pos_automatic_cashdrawer/static/src/js/devices.js

```diff
@@ -0,0 +1,299 @@
+        */
+        automatic_cashdrawer_get_amount_accepted: function () {
+            var done = this.message('cashlogy/get_amount_accepted');
+            // Silently fail
+            done.fail(function (error) { 
+                console.error(error); 
+            });
+            return done;
+        },
+
+        /*
+            Stops acceptance of money
+            @returns 0.00
+        */
+        automatic_cashdrawer_stop_acceptance: function () {
+            var self = this;
+            var done = this.message('cashlogy/stop_acceptance');
+            done.fail(function (error) {
+                var message = error ? error.data.message : _t('Cashdrawer not connected');
+                var body = error ? error.data.debug : _t('Make sure Cashdrawer connected with IOT Box');
+                self.pos.gui.show_popup('error-traceback', {
+                    'title': _t('Cashdrawer Error: ') + message,
+                    'body': body
+                });
+            });
+            return done;
+        },
+
+        /*
+            Gets the inventory of the machine
+            Returns {total: {}, recycler: {}, stacker: {}}
+        */
+        automatic_cashdrawer_get_inventory: function () {
+            var self = this;
+            var done = this.message('cashlogy/get_inventory');
+            done.fail(function (error) {
+                var message = error ? error.data.message : _t('Cashdrawer not connected');
+                var body = error ? error.data.debug : _t('Make sure Cashdrawer connected with IOT Box');
+                self.pos.gui.show_popup('error-traceback', {
+                    'title': _t('Cashdrawer Error: ') + message,
+                    'body': body
+                });
+            });
+            return done;
+        },
+        
+        /*
+            Gets the total amount on the machine
+            Returns {total: 0.00, recycler: 0.00, stacker: 0.00}
+        */
+        automatic_cashdrawer_get_total_amount: function () {
+            var self = this;
+            var done = this.message('cashlogy/get_total_amount');
+            done.fail(function (error) {
+                var message = error ? error.data.message : _t('Cashdrawer not connected');
+                var body = error ? error.data.debug : _t('Make sure Cashdrawer connected with IOT Box');
+                self.pos.gui.show_popup('error-traceback', {
+                    'title': _t('Cashdrawer Error: ') + message,
+                    'body': body
+                });
+            });
+            return done;
+        },
+
+        /*
+            Prints the current inventory of the machine
+        */
+        automatic_cashdrawer_print_inventory: function () {
+            this.pos.gui.show_popup('error', {
+                title: _t('Not implemented'),
+                body: _t('Method not implemented yet')
+            });
+            return $.Deferred().reject('Not implemented yet. TODO');
+        },
```

Comment:
Why use the function if not implemented ?

Reply:
I don't know ;)